homeassistant/components/sensor/__init__.py

```diff
@@ -715,6 +730,17 @@ def state(self) -> Any:
                     f"but provides state {value}:{type(value)} resulting in '{err}'"
                 ) from err
 
+        # Enum checks
```

Comment:
I think that it would be good to provide multiple base classes for the SensorEntity. So this logic would move to a 

`EnumSensorEntity` for which `options` is a required attribute but no unit conversion logic exists.

Reply:
I've played with that implementation after is was quickly discussed earlier. While I think from this perspective, it makes sense, however, when implementing integrations, it will become less "beautiful".

When implementing this on an integration level adds an additional complexity. Right now most integration implements a tuple/list of entity descriptions, iterates over ghat and spawns sensor entities. This is al pretty flexible and straight forward.

Let's assume we split into an `EnumSensorEntity`, and later a `TextEntity`, and maybe even `NumericEntity` and `RangeEntity`. The entity descriptions need to be split as well in a similar fashion.

Now the implementation at the integration level becomes messier. An integration that needs more of these types now all of sudden needs to either create multiple lists and loop over multiple of them and implement multiple entity classes to inherit from their integration sensor class; or use different descriptions (or mixes them and later use `isinstance()` or an added property to their descriptions) to figure out which of their entity sensor subclasses they need.

While adding options in its own subclass makes sense in this place on the code, it will force the implementation down the line to become a little more complex. I'm not sure if that added complexity is worth it.

Reply:
I discussed it with @MartinHjelmare & @emontnemery today, we came to an agreement not to split this any further, to keep the integration implementation less complex.

I'm going to add more validation to avoid weirdness.